### PR TITLE
Add remaining stations endpoints

### PR DESCRIPTION
**Pull request by wasabigeek**
Pending a "delete!" method on Station.

Next step, add measurements.

### CODE REVIEW

File: lib/open_weather/endpoints/stations.rb [hunk 1]
```diff
@@ -10,6 +10,19 @@ def register_station(options = {})
...
+        OpenWeather::Models::Station.new(get("stations/#{id}"))
+      end
+
+      def update_station(id, options = {})
+        OpenWeather::Models::Station.new(put("stations/#{id}", options))
+      end
+
+      def delete_station(id)
+        delete("stations/#{id}")
+        nil
```
Comment (wasabigeek) [inline lib/open_weather/endpoints/stations.rb:24]: @dblock what would you suggest to return here? Currently it the `delete` method returns as empty string (`response.body`).

File: lib/open_weather/endpoints/stations.rb [hunk 2]
```diff
@@ -10,6 +10,19 @@ def register_station(options = {})
       def list_stations
         get('stations').map { |data| OpenWeather::Models::Station.new(data) }
       end
+
+      def get_station(id)
+        OpenWeather::Models::Station.new(get("stations/#{id}"))
+      end
+
+      def update_station(id, options = {})
+        OpenWeather::Models::Station.new(put("stations/#{id}", options))
```
Comment (wasabigeek) [inline lib/open_weather/endpoints/stations.rb:19]: I'm not sure why, but it seems like the API requires `external_id` to be specified 🤷

Comment (wasabigeek) [reply lib/open_weather/endpoints/stations.rb:19]: Also, I could potentially raise an error if `id` was not defined, do you have a suggestion on how to define / where to keep errors?

File: README.md [hunk 3]
```diff
@@ -222,6 +225,33 @@ To list all stations, you can call the client method:
...
+
+#### Update Station
+
+To update a station, you can call the client method:
+```ruby
+data = client.update_station('5ed2118acca8ce0001f1aeg1', external_id: 'SF_TEST002') # => OpenWeather::Models::Station
+```
+Alternatively, call `update!` on an instance of `Station`:
+```ruby
+model = OpenWeather::Models::Station.new(external_id: 'SF_TEST001', ...).register!
```
Comment (dblock) [inline README.md:243]: Put `model.register!` on its own line.

File: lib/open_weather/endpoints/stations.rb (hunk 2 above)
Comment (dblock) [reply lib/open_weather/endpoints/stations.rb:19]: If a parameter is missing, raise ArgumentError, generally I don't specialize errors unless they can happen on all APIs (e.g. incorrect HTTP proxy)

File: lib/open_weather/endpoints/stations.rb (hunk 1 above)
Comment (dblock) [reply lib/open_weather/endpoints/stations.rb:24]: I think `nil` is fine.

File: spec/open_weather/endpoints/stations_spec.rb [hunk 4]
```diff
@@ -42,4 +42,38 @@
       )
     end
   end
+
+  describe '#get_station' do
```
Comment (dblock) [inline spec/open_weather/endpoints/stations_spec.rb:46]: should this be `context`?

Comment (wasabigeek) [reply spec/open_weather/endpoints/stations_spec.rb:46]: Oh I’ve been following this convention http://www.betterspecs.org/#describe (describe for the method under test, context for a scenario)

Comment (dblock) [reply spec/open_weather/endpoints/stations_spec.rb:46]: I've been schooled ;)

Comment (wasabigeek) [reply spec/open_weather/endpoints/stations_spec.rb:46]: They’re aliases so all good either way!

**Comment by dblock** [comment]
Merged, nice work! 

Want to help out with gem maintenance? Make the next release? Drop me an email with your rubygems username/email to dblock at dblock dot org.

**Pull request merged**